### PR TITLE
ENH: MPS Upgrade

### PR DESCRIPTION
**Pull request by ZLLentz**

### CODE REVIEW

File: lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU [hunk 1]
```diff
@@ -12,9 +12,16 @@ VAR_IN_OUT
     fbArbiter: FB_Arbiter;
     fbFFHWO: FB_HardwareFFOutput;
 END_VAR
+VAR_INPUT
+    stFullBeam: ST_BeamParams := PMPS_GVL.cstFullBeam;
```
Comment (ghalym) [inline lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU:16]: can we call it like Maximum Allowed Beam? Just so we don't confuse it with FullBeam.

**Review by ghalym — APPROVED** [review]

File: lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU (hunk 1 above)
Comment (ZLLentz) [reply lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU:16]: I agree with a rename, but that name is not quite right either. We're actually defining a minimum beam setting at which a state is configured safe enough for the FFO to be automatically reset, not a maximum allowed beam.

Comment (slacAWallace) [reply lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU:16]: PossibleMaxBeam? MaxBeamPossible?

File: lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU [hunk 2]
```diff
@@ -54,32 +57,51 @@ END_FOR]]></ST>
...
+ffZeroRate(
+    i_xOk := stBeamNeeded.nRate > 0,
+    i_xAutoReset := TRUE,
+    i_DevName := stMotionStage.sName,
+    i_Desc := 'Device requesting zero rate',
+    i_TypeCode := 16#1001,
+    io_fbFFHWO := fbFFHWO);
+
+// Trip the beam for BPTM timeouts if we want to move
+ffBPTMTimeoutAndMove.i_xReset S= bAtSafeState;
```
Comment (slacAWallace) [inline lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU:95]: I forget does this reset internally?

**Review by slacAWallace — CHANGES_REQUESTED** [review]
Looks like what we discussed.

I would suggest double checking the logic of the resets they may provide more rising edges that you'd like.

File: lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU (hunk 2 above)
Comment (ZLLentz) [reply lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU:95]: I do not understand the question.

File: lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU (hunk 1 above)
Comment (ZLLentz) [reply lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU:16]: `stMaxBeamThreshold` is what I'm going with

File: lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU (hunk 2 above)
Comment (ZLLentz) [reply lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_PositionStatePMPS.TcPOU:95]: Some investigation shows that i_xReset is reset at the HWO end, not at FB_FastFault, so this needs to be adjusted a bit

**Pull request merged**